telegram/ext/conversationhandler.py

```diff
@@ -594,7 +594,8 @@ def update_state(self, new_state: object, key: Tuple[int, ...]) -> None:
         elif new_state is not None:
             if new_state not in self.states:
                 warnings.warn(
-                    "Handler returned state which is unknown to the ConversationHandler."
+                    f"Handler returned state {new_state} which is unknown to the "
+                    f"ConversationHandler {self.name if self.name is not None else ''}."
```

Comment:
Just one very minor nitpick left: when the CH has no name, there will be a trailing whitespace. We can do better, even in warnings :D 

After this, I promise I'll merge! 😉